### PR TITLE
feat: Earthly cache watcher utility

### LINKED ISSUE #267: 🛠️ [TASK] : Make simple cache size watcher for Earthly Cache
### Summary

Make a python utility project called "EarthlyCacheWatchdog".  Look at the standrd python project and example, copy it and set it up.

### Description

Use this library: https://pypi.org/project/watchdog/

We need several parameters:

1. Directory to watch
2. Large File Size (On start any file thats bigger than this gets logged, any file that grows bigger than this gets logged).
3. Max Cache Size (On start or if the cache grows bigger than this, log it).
4. Time window and a max growth size.
    If more than XX (size) grows in the time windows log it.

Use: https://github.com/systemd/python-systemd/
for logging.

It needs to be a systemd service:
https://github.com/torfsen/python-systemd-tutorial

Put this tool in `catalyst-ci/utilities/cache-watchdog`

### PR DESCRIPTION
**Pull request by apskhem**
# Description

Earthly cache watcher to detect large file growth in a specific directory.

## Related Issue(s)

Closes #267

## Description of Changes

* Watch files changes in a specified directory.
* Trigger events when a whether an individual file or a watched directory grows exceeding the certain criteria.
* Main triggering criteria: single file size exceeds, watched directory size exceeds, watched directory growth in size within an interval exceeds.

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by minikin** [comment]
@stevenj can you review, please?

**Review by bkioshn — COMMENTED** [review]
Looks great, just minor wording improvement 🥳

File: utilities/earthly-cache-watcher/helper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    
+    relative_path = os.path.relpath(path, working_dir_path)
+    parts = relative_path.split(os.sep)
+    
+    if parts:
+        return parts[0]
+    return None
+
+def add_or_init(obj: dict[str, int], key: str, value: int):
+    if key not in obj:
```
Comment (Mr-Leshiy) [inline utilities/earthly-cache-watcher/helper.py:44]: you can use `setdefault` method of the dict

**Review by kukkok3 — APPROVED** [review]
LGTM

**Review by kukkok3 — APPROVED** [review]
LGTM

**Pull request merged**